lib/tools/adb-emu-commands.js

```diff
@@ -36,6 +36,25 @@ emuMethods.rotate = async function () {
   await this.adbExec(['emu', 'rotate']);
 };
 
+emuMethods.sendSMS = async function (phoneNumber, message) {
+  if (_.isUndefined(phoneNumber)) {
+    log.errorAndThrow('sendSMS phoneNumber param is undefined');
+  }
+  phoneNumber = (phoneNumber+"").replace(/\s*/, "");
+  if (!/^[\+]?[(]?[0-9]*[)]?[-\s\.]?[0-9]*[-\s\.]?[0-9]{2,}$/im.test(phoneNumber)) {
+    log.errorAndThrow(`Invalid phoneNumber param ${phoneNumber}`);
+  }
+
+  if (_.isUndefined(message) && _.isEmpty(message)) {
```

Comment:
Better to give a default and skip the `undefined` check:
```js
emuMethods.sendSMS = async function (phoneNumber, message = '') {
  // ...
  if (message === '') {
    log.errorAndThrow('Sending an SMS requires a message');
  }
  // ...
}
```

This could also go at the very top, before processing anything.